src/components/RouletteMain/RouletteMain.jsx

```diff
@@ -52,53 +60,39 @@ export default function RouletteMain() {
     }
 
     function onAddMovieSubmitAction(item) {
-        let stateList = [...movieList];
-        let uniqueId = stateList.length + 1;
-        item.id = uniqueId;
-        stateList.push(item);
-        setMovieList(stateList)
-        filterMovieVisibility(stateList);
+        props.addMovie(item)
         onCloseAction();
     }
 
     function onEditMovieSaveAction(item) {
-        let stateList = [...movieList];
-        stateList.forEach((obj, i) => { if (obj.id === item.id) { stateList[i] = item } });
-        setMovieList(stateList)
-        filterMovieVisibility(stateList);
+        props.editMovie(item)
         onCloseAction();
     }
 
     function onDeleteMovieConfirmAction(item) {
-        let stateList = [...movieList];
-        stateList = stateList.filter((obj) => obj.id !== item.id);
-        setMovieList(stateList);
-        setMovieList((state) => {
-            console.log(state);
-            filterMovieVisibility(state);
-            onCloseAction();
-            return state;
-          }); 
+        props.deleteMovie(item.id);
+        onCloseAction();
     }
 
     function onShowMovieDetailsAction(id) {
-        let stateList = [...movieList];
+        props.getMovie(id)
+        let stateList = [...props.movieProps];
         let listObject = stateList.find(obj => obj.id === id);
         setMovieDetailSelection(listObject);
         setMovieDetailsVisible(true);
         window.scrollTo(0, 0);
     }
 
-    function onFilterCategoryClicked(category) {       
-        let stateList = sortByKey([...movieList]);
+    function onFilterCategoryClicked(category) {
         setFilterOption(category);
-        filterMovieVisibility(stateList, category);
+        props.filterMovies(category);
     }
 
     function onHandleSelectedSortOption(selectedOption) {
         setSortOption(selectedOption);
-        let stateList = sortByKey(visibleMovies, selectedOption);
-        filterMovieVisibility(stateList);
+        setSortOption((state) => {
+            props.sortMovies([...props.movieProps], state)
```

Comment:
You do not need send movies to the action and reducer. One of the way how to implement this is to update sort value in the store and sort movies in the **mapStateToProps** function. Another possible solution based on the fact that redux-thunk provides **getState** function for the action and you can get access to the store in the action function:
```
function sortMovies(array, key) {
    return (dispatch, getState) => {
        const {
            movieReducer
        } = getState();
        let sortedArray = sortByKey([...movieReducer], key);
                dispatch({
                    type: SORT_MOVIES,
                    payload: sortedArray
                })
            }
}
```